From 8d7a5b0eaa5ad40daa54e68b1fa6e872af142d20 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:12:35 -0400
Subject: [ethernet] igb: do a reset on SR-IOV re-init if device is down

Message-id: <1409062359-13063-78-git-send-email-sassmann@redhat.com>
Patchwork-id: 89181
O-Subject: [RHEL7.1 PATCH 77/81] igb: do a reset on SR-IOV re-init if device is down
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Stefan Assmann <sassmann@kpanic.de>

To properly re-initialize SR-IOV it is necessary to reset the device
even if it is already down. Not doing this may result in Tx unit hangs.

Cc: stable <stable@vger.kernel.org>
Signed-off-by: Stefan Assmann <sassmann@kpanic.de>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 76252723e88681628a3dbb9c09c963e095476f73)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_main.c b/drivers/net/ethernet/intel/igb/igb_main.c
index 261e2d4..862c97c 100644
--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -7594,6 +7594,8 @@ static int igb_sriov_reinit(struct pci_dev *dev)
 
  if (netif_running(netdev))
   igb_close(netdev);
+ else
+  igb_reset(adapter);
 
  igb_clear_interrupt_scheme(adapter);
 
-- 
1.7.1